src/main/java/frc/robot/commands/CollectThatCone.java

```diff
@@ -0,0 +1,75 @@
+package frc.robot.commands;
+
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.geometry.Transform2d;
+import edu.wpi.first.math.geometry.Translation2d;
+import edu.wpi.first.math.util.Units;
+import edu.wpi.first.wpilibj2.command.ParallelDeadlineGroup;
+import edu.wpi.first.wpilibj2.command.RunCommand;
+import edu.wpi.first.wpilibj2.command.SequentialCommandGroup;
+import frc.robot.subsystems.Intake;
+import frc.robot.subsystems.IntakeLimelight;
+import frc.robot.subsystems.IntakeLimelight.ConeDetection;
+import frc.robot.subsystems.Lift;
+import frc.robot.subsystems.Lights;
+import frc.robot.subsystems.drive.DriveSubsystem;
+import java.util.Optional;
+
+public class CollectThatCone extends SequentialCommandGroup {
+
+  private static final double DISTANCE_IN_FRONT_OF_CONE_METERS = Units.inchesToMeters(3);
+  private Optional<ConeDetection> coneDetection = Optional.empty();
+  private Pose2d inFrontOfCone = new Pose2d();
+  private static final double SPEED_AT_CONE_MPS = 2.0;
+
+  private static Pose2d getPoseInFrontOfCone(DriveSubsystem drive, ConeDetection cone) {
+    Rotation2d coneAngle = Rotation2d.fromDegrees(cone.angleDeg);
+    double coneDistanceMeters = cone.distanceMeters + Units.inchesToMeters(6);
+    Pose2d poseAtDetection = drive.getPastPose(cone.latencySec);
+    Transform2d toInFrontOfCone =
+        new Transform2d(
+            new Translation2d(coneDistanceMeters - DISTANCE_IN_FRONT_OF_CONE_METERS, 0)
+                .rotateBy(coneAngle),
+            coneAngle);
+    return poseAtDetection.plus(toInFrontOfCone);
+  }
+
+  public CollectThatCone(
+      DriveSubsystem drive, IntakeLimelight limelight, Intake intake, Lift lift, Lights lights) {
+    addRequirements(drive, intake, lift);
```

Comment:
Kind of a similar questions as earlier, why do only some subsystems need to be passed in as requirements? Why not limelight and lights?

Reply:
Limelight doesn't even really need to be a subsystem honestly, the question is whether we want to guarantee that only one command is using the subsystem at a time. For lights and limelight, we don't really care. For something like lift, it can only go to one position at a time, so we wouldn't want multiple commands sending positions to go to